### PR TITLE
add auto_merge parameter to deploy

### LINKED ISSUE #39: 409 on deploy
Howdy, maybe you could give me a clue. I do `bot deploy v-test to stage`.

and get a strange response such as bellow:
`[Thu Mar 18 2021 18:20:56 GMT+0000 (Coordinated Universal Time)] ERROR 409 Conflict merging dev into v-test`

dev it's my default branch while v-test is a tag... Not very close with gh deployments and its API, but I'm definitely misleaded by this response...

### PR DESCRIPTION
**Pull request by dennybaa**
Fixes #39

### CODE REVIEW

File: src/github-deployments.coffee [hunk 1]
```diff
@@ -8,8 +8,10 @@
 #   HUBOT_GITHUB_TOKEN - GitHub API token. Required to perform authenticated actions.
 #   HUBOT_GITHUB_API - (optional) The base API URL. This is useful for Enterprise Github installations.
 #   HUBOT_GITHUB_USER - Default GitHub username to use if one is not given.
-#   HUBOT_GITHUB_DEPLOY_TARGETS - comma separated keys for your deployment environments.
 #   HUBOT_GITHUB_REPO - GitHub repository to use for deployments
+#   HUBOT_GITHUB_DEPLOY_TARGETS - comma separated keys for your deployment environments.
+#   HUBOT_GITHUB_DEPLOY_AUTO_MERGE - (optional) Instructs GitHub to attempt to automatically merge the default branch into the requested ref.
+#   HUBOT_GITHUB_DEPLOY_ALLOW_CHANNEL_IDS - (optional) Specifies list of channel IDs allowed to invoke deployments.
```
Comment (stephenyeargin) [inline src/github-deployments.coffee:14]: Looks like this is unrelated.

Comment (stephenyeargin) [inline src/github-deployments.coffee]: Looks like this is unrelated.

```suggestion
```

File: src/github-deployments.coffee [hunk 2]
```diff
@@ -33,19 +35,20 @@ module.exports = (robot) ->
...
   # Status
-  robot.respond /deploy status( [0-9]+)?(?: for ([\-A-z0-9]+)\/([\-A-z0-9]+))?$/i, (msg) ->
-    unless checkConfiguration(msg)
+  robot.respond /deploy status( [0-9]+)?(?: for ([\-A-z0-9]+)\/([\-A-z0-9]+))?$/i, (res) ->
+    unless checkConfiguration(res)
       return;
 
     app = process.env.HUBOT_GITHUB_REPO
-    status_id = msg.match[1]
+    channel_ids = process.env.HUBOT_GITHUB_DEPLOY_ALLOW_CHANNEL_IDS.split(",")
```
Comment (stephenyeargin) [inline src/github-deployments.coffee]: Same as above. It's an interesting idea for a feature, but we'd want to keep it separate and add test coverage around it.

```suggestion
```

**Review by stephenyeargin — CHANGES_REQUESTED** [review]
Couple of requested changes to make the tests pass.

I'm also wondering if we should make the auto-merge option part of the deployment command rather than have it as a configuration variable. Reason being that I can see that most users would want the default behavior maintained, but may have a special proof of concept branch that doesn't need to be merged in.

If it were a CLI command, I'd say add something like `--no-auto-merge` or something. We'd likely introduce some complexity in the RegEx (and want test cases) to support additional arguments.

File: src/github-deployments.coffee (hunk 2 above)
Comment (dennybaa) [reply src/github-deployments.coffee]: yup this is not finished yet

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "hubot-github-deployments",
   "description": "Integrate with GitHub deployment API",
-  "version": "1.2.3",
+  "version": "1.2.4",
```
Comment (stephenyeargin) [inline package.json]: No need to bump the version here. We're using [`release-it`](https://github.com/release-it/release-it) to handle that process. 

```suggestion
  "version": "1.2.3",
```

**Comment by dennybaa** [comment]
@stephenyeargin not sure about all, I use tags... Definitely will work as a workaround for now. Especial if you release previous versions (tags for v1.2.x and current might be v1.3.x). Moreover the deployment API action doesn't merge anything... This default behavior is very opinionated.

**Comment by dennybaa** [comment]
> If it were a CLI command, I'd say add something like --no-auto-merge or something. We'd likely introduce some complexity in the RegEx (and want test cases) to support additional arguments.

Adding to CLI is an overkill imho at this point. When there are cases or requests then will make sense...

**Comment by stephenyeargin** [comment]
> @stephenyeargin not sure about all, I use tags... Definitely will work as a workaround for now. Especial if you release previous versions (tags for v1.2.x and current might be v1.3.x). Moreover the deployment API action doesn't merge anything... This default behavior is very opinionated.

Good thought. I'll go with this.

**Review by stephenyeargin — APPROVED** [review]

**Comment by stephenyeargin** [comment]
Released as part of `v1.2.4` :tada:

**Pull request merged**